### ISSUE TITLE
`no-redundant-roles` does not detect nullish attributes

### ISSUE THREAD

**Issue by ouuan** [vue-a11y/eslint-plugin-vuejs-accessibility#1321]
For example, I have an `<a>` tag that is sometimes disabled:

```vue
<a
  :href="disabled ? undefined : 'https://example.com'"
  :aria-disabled="disabled"
  role="link"
>
  maybe disabled
</a>
```

The w3c specification requires a `role="link"` when it the link is disabled: https://www.w3.org/TR/html-aria/#example-communicate-a-disabled-link-with-aria

But now this plugin does not know that the `href` is sometimes not set and reports an error for `no-redundant-roles`.

I'm not familiar with ESLint plugin development, but I'm wondering if this can be fixed by using type information and checking whether the attribute type can be nullish.
